cvelib/cli.py

```diff
@@ -298,32 +301,54 @@ def publish(ctx: click.Context, cve_id: str, cve_json_str: str, print_raw: bool)
 def reject(ctx: click.Context, cve_id: str, cve_json_str: str, print_raw: bool) -> None:
     """Reject a CVE record for a reserved or published CVE ID.
 
-    Will update if the CVE record already exists.
+    If the CVE is already rejected, this action will update its record if one is supplied.
+    A rejected CVE with a record can only be moved to the published state (see `cve publish`). A
+    rejected CVE without a record can be moved to the reserved state.
+
+    Example:
 
     \b
-    cve reject 'CVE-2022-1234' --json '{"rejectedReasons": [{"lang": "en", "value": "A reason."}]}'
+    cve reject CVE-2022-1234 -j '{"rejectedReasons": [{"lang": "en", "value": "A reason."}]}'
 
     For information on the required properties in a given CVE JSON record, see the
     `cnaRejectedContainer` schema in:\n
     https://github.com/CVEProject/cve-schema/blob/master/schema/v5.0/CVE_JSON_5.0_schema.json
     """
-    try:
-        cve_json = json.loads(cve_json_str)
-    except json.JSONDecodeError as exc:
-        click.echo("CVE data was not valid JSON. Error was:\n")
-        click.secho(str(exc))
-        return
+    cve_json = None
+    if cve_json_str:
+        try:
+            cve_json = json.loads(cve_json_str)
+        except json.JSONDecodeError as exc:
+            click.echo("CVE data was not valid JSON. Error was:\n")
+            click.secho(str(exc))
+            return
+
     if ctx.obj.interactive:
         click.echo("You are about to reject ", nl=False)
         click.secho(cve_id, bold=True, nl=False)
-        click.echo(" using the following input:\n\n", nl=False)
-        click.secho(cve_json_str, bold=True, nl=False)
-        if not click.confirm("\n\nThis operation cannot be reversed; do you want to continue?"):
+        if cve_json:
+            click.echo(" using the following input:\n\n", nl=False)
+            click.secho(cve_json_str)
+        else:
+            click.echo(" without providing a reject record.")
+        if not click.confirm("\nDo you want to continue?"):
             click.echo("Exiting...")
             sys.exit(0)
         click.echo()
 
     cve_api = ctx.obj.cve_api
+
+    # Reject a CVE ID without a record
+    if cve_json is None:
+        response_data = cve_api.move_to_rejected(cve_id)
+        if print_raw:
+            print_json_data(response_data)
+        else:
+            click.echo("Rejected the following CVE:\n")
+            print_cve_id(response_data["updated"])
```

Comment:
response_data["updated"] is what we get when updating a record, and we pass that to print_cve_record. If we're rejecting without a record, and calling print_cve_id, do we need to pass something else?

Reply:
The response still includes the updated CVE ID. And the `updated` vs `created` is based on whether you call PUT or POST, not reflecting an update/create of a record:
```
> cve reject CVE-2022-35619 --raw
{
    "message": "CVE-2022-35619 was successfully updated.",
    "updated": {
        "cve_id": "CVE-2022-35619",
        "cve_year": "2022",
        "owning_cna": "redhat",
        "requested_by": {
            "cna": "redhat",
            "user": "mprpic@redhat.com"
        },
        "reserved": "2022-09-27T17:42:37.366Z",
        "state": "REJECTED",
        "time": {
            "created": "2022-09-27T17:42:37.366Z",
            "modified": "2022-09-30T15:19:54.332Z"
        }
    }
}
```

Reply:
Ack, thanks for confirming! I couldn't remember what data the /cve-id endpoint returned vs. the /cve endpoint.